lib/cc/engine/analyzers/reporter.rb

```diff
@@ -72,13 +72,18 @@ def process_files
           processed_files_count = Concurrent::AtomicFixnum.new
 
           pool.run do |file|
-            CC.logger.debug("Processing #{lang} file: #{file}")
+            begin
+              CC.logger.debug("Processing #{lang} file: #{file}")
 
-            sexp = language_strategy.run(file)
+              sexp = language_strategy.run(file)
 
-            process_sexp(sexp)
+              process_sexp(sexp)
 
-            processed_files_count.increment
+              processed_files_count.increment
+            rescue Exception => ex
+              CC.logger.warn("Error processing file: #{file}")
```

Comment:
there is no existing spec code for the reporter, IMO this change is small enough where it's not worth setting that up.

i did verify this with a problematic repo locally:
```
codeclimate-parser socket not present
waiting 1s...
W, [2017-11-14T22:22:03.709182 #1]  WARN -- : Error processing file: path/to/file.rb
/var/lib/gems/2.3.0/gems/sexp_processor-4.10.0/lib/sexp.rb:57:in `new': stack level too deep (SystemStackError)
```